src/pycram/external_interfaces/robokudo.py

```diff
@@ -1,135 +1,259 @@
 
-    return obj_msg
+    global query_result
 
+    def active_callback():
+        rospy.loginfo("Send query to RoboKudo to scan a specific region")
 
-def make_query_goal_msg(obj_desc: ObjectDesignatorDescription) -> 'QueryGoal':
+    def done_callback(state, result):
+        rospy.loginfo("Finished perceiving")
+        global query_result
+        query_result = result.res
+
+    def feedback_callback(msg):
+        pass
+
+    object_goal = QueryGoal()
+    object_goal.obj.location = str(region)
+    client = actionlib.SimpleActionClient('robokudo/query', QueryAction)
+    rospy.loginfo("Waiting for action server")
+    client.wait_for_server()
+    client.send_goal(object_goal, active_cb=active_callback, done_cb=done_callback, feedback_cb=feedback_callback)
+    client.wait_for_result()
+
+    return query_result
+
+
+def query_human() -> PointStamped:
     """
-    Creates a QueryGoal message from a PyCRAM Object designator description for the use of Querying RobotKudo.
+    Sends a query to RoboKudo to locate a human and returns a PointStamped object representing the pose where the human is located.
+    Continuously publishes the pose onto the topic /human_pose.
 
-    :param obj_desc: The PyCRAM object designator description that should be converted
-    :return: The RoboKudo QueryGoal for the given object designator description
+    :return: A PointStamped object containing the pose of the detected human.
     """
-    goal_msg = QueryGoal()
-    goal_msg.obj.uid = str(id(obj_desc))
-    goal_msg.obj.obj_type = str(obj_desc.types[0].name)  # For testing purposes
-    if ObjectType.JEROEN_CUP == obj_desc.types[0]:
-        goal_msg.obj.color.append("blue")
-    elif ObjectType.BOWL == obj_desc.types[0]:
-        goal_msg.obj.color.append("red")
-    return goal_msg
+    init_robokudo_interface()
+    from robokudo_msgs.msg import QueryAction, QueryGoal, QueryResult
 
+    global human_bool
+    global query_result
+    global human_pose
 
-@init_robokudo_interface
-def query(object_desc: ObjectDesignatorDescription) -> ObjectDesignatorDescription.Object:
+    def active_callback():
+        rospy.loginfo("Send query to RoboKudo to perceive a human")
+
+    def done_callback(state, result):
+        rospy.loginfo("Finished perceiving")
+        global human_bool
+        human_bool = True
+        global query_result
+        query_result = result
+
+    def feedback_callback(msg):
+        global feedback_result
+        feedback_result = msg
+
+    def callback(pose):
+        global human_bool
+        global human_pose
+        human_bool = True
+        human_pose = pose
+
+    client = actionlib.SimpleActionClient('robokudo/query', QueryAction)
+    rospy.loginfo("Waiting for action server")
+    client.wait_for_server()
+    object_goal = QueryGoal()
+    object_goal.obj.type = 'human'
+    client.send_goal(object_goal, active_cb=active_callback, done_cb=done_callback, feedback_cb=feedback_callback)
```

Comment:
The code for the action client is in every method, it would be nice to move this to a seperate function which handles the action client.

Reply:
code is wip freezing this PR for time being till its done